### PR TITLE
Require setup to be completed for all users to access dashboard

### PR DESCRIPTION
**Pull request by aaemnnosttv**
## Summary

Addresses issue #352

## Relevant technical choices

* Updates broken E2E tests

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: tests/e2e/specs/admin-tracking.test.js [hunk 1]
```diff
@@ -11,17 +6,17 @@ import { activatePlugin, deactivatePlugin, visitAdminPage } from '@wordpress/e2e
...
+describe( 'management of tracking opt-in/out via settings page', () => {
 
 	beforeAll( async() => {
 		await activatePlugin( 'e2e-tests-auth-plugin' );
 		await activatePlugin( 'e2e-tests-site-verification-plugin' );
 	} );
 
 	beforeEach( async() => {
-		await resetSiteKit();
+		await setSearchConsoleProperty();
```
Comment (felixarntz) [inline tests/e2e/specs/admin-tracking.test.js]: Wouldn't it be better to have this work through a utility plugin? It's the same type of setup requirement like being authenticated and verified, which also have plugins.

File: tests/e2e/specs/admin-tracking.test.js [hunk 2]
```diff
@@ -39,77 +38,56 @@ describe( 'Providing client configuration', () => {
...
-		);
-		expect( analyticsScriptTag.length ).toEqual( 0 );
+		await expect( page ).not.toMatchElement( 'script[src^="https://www.google-analytics.com/analytics.js"]' );
 
 		// Ensure no tag manager script exists.
-		const tagManagerScriptTag = await page.$x(
-			'//script[contains(@src,"https://www.googletagmanager.com/gtag/js?id=UA-130569087-3")]'
-		);
-		expect( tagManagerScriptTag.length ).toEqual( 0 );
+		await expect( page ).not.toMatchElement( 'script[src^="https://www.googletagmanager.com/gtag/js?id=UA-130569087-3"]' );
```
Comment (felixarntz) [inline tests/e2e/specs/admin-tracking.test.js]: Why did you make all these changes to the tracking test? How do they relate to the Search Console property being required for permission?

File: tests/e2e/specs/auth-flow.test.js [hunk 3]
```diff
@@ -34,6 +39,7 @@ describe( 'Site Kit set up flow for the first time', () => {
 
 	beforeAll( async() => {
 		await activatePlugin( 'e2e-tests-oauth-callback-plugin' );
+		await setSearchConsoleProperty();
```
Comment (felixarntz) [inline tests/e2e/specs/auth-flow.test.js]: As mentioned, this is a general prerequisite for a ton of tests (now and future), so it should be a utility plugin I believe. Setting it _during_ a test should only happen in the specific test for Search Console setup.

File: tests/e2e/specs/modules/analytics/setup-with-account-no-tag.test.js [hunk 4]
```diff
@@ -29,13 +35,7 @@ const setReferenceUrl = async() => {
...
 	beforeAll( async() => {
 		await page.setRequestInterception( true );
-		page.on( 'request', request => {
-			if ( ! request._allowInterception ) {
-
-				// prevent errors for requests that happen after interception is disabled.
-				return;
-			}
-
+		useRequestInterception( request => {
```
Comment (felixarntz) [inline tests/e2e/specs/modules/analytics/setup-with-account-no-tag.test.js]: How is this change related?

File: tests/e2e/specs/modules/pagespeed-insights/activation.test.js [hunk 5]
```diff
@@ -6,19 +6,25 @@ import { visitAdminPage, activatePlugin } from '@wordpress/e2e-test-utils';
...
+	resetSiteKit,
+	setSearchConsoleProperty,
+	setSiteVerification,
+} from '../../../utils';
 
 describe( 'PageSpeed Insights Activation', () => {
 	beforeEach( async() => {
 		await activatePlugin( 'e2e-tests-auth-plugin' );
-		await activatePlugin( 'e2e-tests-site-verification-plugin' );
+		await setSiteVerification();
```
Comment (felixarntz) [inline tests/e2e/specs/modules/pagespeed-insights/activation.test.js]: Hmm, you replaced the utility plugin with setting that value via JS - why that? This ties in with my other comments, I think such base requirements should happen via utility plugin, unless for tests in which these actions specifically are tested.

File: tests/e2e/specs/modules/pagespeed-insights/activation.test.js [hunk 6]
```diff
@@ -38,7 +44,7 @@ describe( 'PageSpeed Insights Activation', () => {
 
 		await expect( page ).toMatchElement( 'h2.googlesitekit-setup-module__title', { text: 'PageSpeed Insights' } );
 
-		await page.type( 'input.mdc-text-field__input', 'PSIKEYTOSUBMITANDTEST' );
+		await pasteText( 'input.mdc-text-field__input', 'PSIKEYTOSUBMITANDTEST' );
```
Comment (felixarntz) [inline tests/e2e/specs/modules/pagespeed-insights/activation.test.js]: This also seems unrelated, unless I'm mistaken.

File: tests/e2e/specs/admin-tracking.test.js (hunk 2 above)
Comment (aaemnnosttv) [reply tests/e2e/specs/admin-tracking.test.js]: These were quick changes to bring this test up to date with the patterns and utilities we're using which are mostly for consistency and readability. It isn't directly related to this requirement but we have been updating some of the older E2E tests as we touch them. If you'd prefer this to go through another issue we can back out any non-critical changes.

File: tests/e2e/specs/modules/pagespeed-insights/activation.test.js (hunk 5 above)
Comment (aaemnnosttv) [reply tests/e2e/specs/modules/pagespeed-insights/activation.test.js]: One reason is for speed. Activating utility plugins happens one at a time and is rather slow compared to the API-based utility, especially when running the test over and over in development. Perhaps a utility plugin may be the proper implementation here, but ideally we would have a single plugin for skipping/faking the setup, which we don't (although I suppose we could). Each piece of the setup has been split into separate utility plugins due to the need to activate some pieces while not activating others. Even the auth plugin is doing too much IMO and should probably be split into 2 or 3 pieces.

The setup utility plugins (auth + site verification) are currently filter-based which makes them only applicable to the current user. This makes them unusable for multi-user tests and requires that we have utilities to actually set the data in the DB which is what the REST utilities do. They don't have to be filter-based of course, but then why use a utility plugin for setting data on activation when we have a persistent REST route for doing the same thing while providing much more control.

My understanding and approach recently has been that we were going to leverage the REST based utilities more and use utility plugins for cases which were more test-specific where needed.

There's clearly more than one way to go about this so perhaps we should discuss this a bit more to clarify the direction going forward 😄

File: tests/e2e/specs/modules/pagespeed-insights/activation.test.js (hunk 6 above)
Comment (aaemnnosttv) [reply tests/e2e/specs/modules/pagespeed-insights/activation.test.js]: Correct. We have been updating/enhancing older tests with newer utilities as we come back to them.

File: tests/e2e/specs/modules/analytics/setup-with-account-no-tag.test.js (hunk 4 above)
Comment (aaemnnosttv) [reply tests/e2e/specs/modules/analytics/setup-with-account-no-tag.test.js]: We have been updating/enhancing older tests with newer utilities as we come back to them for consistency.

File: tests/e2e/specs/modules/pagespeed-insights/activation.test.js (hunk 5 above)
Comment (aaemnnosttv) [reply tests/e2e/specs/modules/pagespeed-insights/activation.test.js]: One reason is for speed. Activating utility plugins happens one at a time and is rather slow compared to the API-based utility, especially when running the test over and over in development. Perhaps a utility plugin may be the proper implementation here, but ideally we would have a single plugin for skipping/faking the setup, which we don't (although I suppose we could). Each piece of the setup has been split into separate utility plugins due to the need to activate some pieces while not activating others. Even the auth plugin is doing too much IMO and should probably be split into 2 or 3 pieces.

The setup utility plugins (auth + site verification) are currently filter-based which makes them only applicable to the current user. This makes them unusable for multi-user tests and requires that we have utilities to actually set the data in the DB which is what the REST utilities do. They don't have to be filter-based of course, but then why use a utility plugin for setting data on activation when we have a persistent REST route for doing the same thing while providing much more control.

My understanding and approach recently has been that we were going to leverage the REST based utilities more and use utility plugins for cases which were more test-specific where needed.

There's clearly more than one way to go about this so perhaps we should discuss this a bit more to clarify the direction going forward 😄

File: tests/e2e/specs/admin-tracking.test.js (hunk 1 above)
Comment (aaemnnosttv) [reply tests/e2e/specs/admin-tracking.test.js]: Perhaps - see my comment below.
